nucleus/admin/launcher/src/main/java/com/sun/enterprise/admin/launcher/JvmOptions.java

```diff
@@ -58,9 +58,19 @@
  */
 class JvmOptions {
 
+    /**
+     * Pattern for recognising environment variables which are in the style of ${ENV=foo}
+     * @see org.glassfish.config.support.TranslatedConfigView
+     */
     private final static Pattern envP = Pattern.compile("([^\\$]*)\\$\\{ENV=([^\\}]*)\\}([^\\$]*)");
     private static final int MAX_SUBSTITUTION_DEPTH = 100;
     
+    Map<String, String> sysProps = new HashMap<String, String>();
```

Comment:
You don't need the extra generic parameters here:

    Map<String, String> sysProps = new HashMap<();